src/main/frontend/amplify.js

```diff
@@ -0,0 +1,142 @@
1}return e.prototype.update=function(e){if(this.finished)throw new Error("Attempted to update an already finished hash.");var t=0,r=e.byteLength;if(this.bytesHashed+=r,8*this.bytesHashed>n.MAX_HASHABLE_LENGTH)throw new Error("Cannot hash more than 2^53 - 1 bits");for(;r>0;)this.buffer[this.bufferLength++]=e[t++],r--,this.bufferLength===n.BLOCK_SIZE&&(this.hashBuffer(),this.bufferLength=0)},e.prototype.digest=function(){if(!this.finished){var e=8*this.bytesHashed,t=new DataView(this.buffer.buffer,this.buffer.byteOffset,this.buffer.byteLength),r=this.bufferLength;if(t.setUint8(this.bufferLength++,128),r%n.BLOCK_SIZE>=n.BLOCK_SIZE-8){for(var o=this.bufferLength;o<n.BLOCK_SIZE;o++)t.setUint8(o,0);this.hashBuffer(),this.bufferLength=0}for(o=this.bufferLength;o<n.BLOCK_SIZE-8;o++)t.setUint8(o,0);t.setUint32(n.BLOCK_SIZE-8,Math.floor(e/4294967296),!0),t.setUint32(n.BLOCK_SIZE-4,e),this.hashBuffer(),this.finished=!0}var i=new Uint8Array(n.DIGEST_LENGTH);for(o=0;o<8;o++)i[4*o]=this.state[o]>>>24&255,i[4*o+1]=this.state[o]>>>16&255,i[4*o+2]=this.state[o]>>>8&255,i[4*o+3]=this.state[o]>>>0&255;return i},e.prototype.hashBuffer=function(){for(var e=this.buffer,t=this.state,r=t[0],o=t[1],i=t[2],a=t[3],s=t[4],u=t[5],l=t[6],c=t[7],d=0;d<n.BLOCK_SIZE;d++){if(d<16)this.temp[d]=(255&e[4*d])<<24|(255&e[4*d+1])<<16|(255&e[4*d+2])<<8|255&e[4*d+3];else{var f=this.temp[d-2],h=(f>>>17|f<<15)^(f>>>19|f<<13)^f>>>10,p=((f=this.temp[d-15])>>>7|f<<25)^(f>>>18|f<<14)^f>>>3;this.temp[d]=(h+this.temp[d-7]|0)+(p+this.temp[d-16]|0)}var m=(((s>>>6|s<<26)^(s>>>11|s<<21)^(s>>>25|s<<7))+(s&u^~s&l)|0)+(c+(n.KEY[d]+this.temp[d]|0)|0)|0,g=((r>>>2|r<<30)^(r>>>13|r<<19)^(r>>>22|r<<10))+(r&o^r&i^o&i)|0;c=l,l=u,u=s,s=a+m|0,a=i,i=o,o=r,r=m+g|0}t[0]+=r,t[1]+=o,t[2]+=i,t[3]+=a,t[4]+=s,t[5]+=u,t[6]+=l,t[7]+=c},e}();e.exports.RawSha256=r})),i.register("cI1IX",(function(e,t){"use strict";Object.defineProperty(e.exports,"__esModule",{value:!0}),e.exports.uint32ArrayFrom=e.exports.numToUint8=e.exports.isEmptyData=e.exports.convertToBuffer=void 0;var n=i("9bDQf");Object.defineProperty(e.exports,"convertToBuffer",{enumerable:!0,get:function(){return n.convertToBuffer}});var r=i("1RGZe");Object.defineProperty(e.exports,"isEmptyData",{enumerable:!0,get:function(){return r.isEmptyData}});var o=i("2xCFW");Object.defineProperty(e.exports,"numToUint8",{enumerable:!0,get:function(){return o.numToUint8}});var a=i("11P6Z");Object.defineProperty(e.exports,"uint32ArrayFrom",{enumerable:!0,get:function(){return a.uint32ArrayFrom}})})),i.register("9bDQf",(function(e,t){var n=i("N0o3H").Buffer;Object.defineProperty(e.exports,"__esModule",{value:!0}),e.exports.convertToBuffer=void 0;var r=i("4Zaep"),o=void 0!==n&&n.from?function(e){return n.from(e,"utf8")}:r.fromUtf8;e.exports.convertToBuffer=function(e){return e instanceof Uint8Array?e:"string"==typeof e?o(e):ArrayBuffer.isView(e)?new Uint8Array(e.buffer,e.byteOffset,e.byteLength/Uint8Array.BYTES_PER_ELEMENT):new Uint8Array(e)}})),i.register("N0o3H",(function(t,n){
```

Comment:
Until we have yarn workspaces, could be nice to have this generated by a build step. tldraw does this with [this script](https://github.com/logseq/logseq/blob/refactor/fs2/tldraw/apps/tldraw-logseq/build.mjs) and then hooking [in here](https://github.com/logseq/logseq/blob/7021b5a040ec0301027b74b0c1751a5b813719d1/package.json#L74-L75)

Reply:
Using npm hooks to pre-build dependency libraries is a great practice. However, the amplify/* library may not be frequently updated in the future, so I temporarily placed the build package separately in the resources/* directory to avoid taking up space in main.js. If there are frequent updates in the future, npm hooks can be used again.